federation-js/src/composition/__tests__/composeAndValidate.test.ts

```diff
@@ -614,6 +669,16 @@ describe('composition of value types', () => {
       expect(compositionResult.errors[0]).toMatchInlineSnapshot(`
         Object {
           "code": "VALUE_TYPE_UNION_TYPES_MISMATCH",
+          "locations": Array [
+            Object {
+              "column": 1,
+              "line": 14,
+            },
+            Object {
+              "column": 1,
+              "line": 14,
+            },
+          ],
```

Comment:
Should these locations point to the specific types within the union that are in violation? It looks like we identify that in the error message so I think that info is available to us.

Something just crossed my mind. How does a person (or code) identify how to map a location in this array back to the correct service? Do we need to append some additional info to the error's `extensions` in order to provide that mapping?

Reply:
Yes definitely - I think since Martijn brought this up in an earlier comment it seems like this will be covered in the future [when we add support for adding service names to composition errors.](https://github.com/apollographql/federation/issues/705) We could have structured data with locations associated to service name.

Reply:
To answer the first question about the union types. Since right now we don't have a way of specifying which locations belong to which services, we won't know which of the possibly numerous locations that would re reported back here correspond to the service name. Could I add a [comment here to come back and fix this as part of this issue in the future?](https://github.com/apollographql/federation/issues/705)

Reply:
Yes that sounds good to me! If you're collecting these in a single issue, tagging these TODOs for searchability would be nice, i.e. `TODO(<issue #>)`